config/initializers/warden.rb

```diff
@@ -0,0 +1,22 @@
+# Strategies provided by OpenProject:
+require 'warden/strategies/basic_auth_failure'
+require 'warden/strategies/global_basic_auth'
+require 'warden/strategies/user_basic_auth'
+require 'warden/strategies/session'
+
+strategies = [
+  [:basic_auth_failure, Warden::Strategies::BasicAuthFailure],
+  [:global_basic_auth,  Warden::Strategies::GlobalBasicAuth],
+  [:user_basic_auth,    Warden::Strategies::UserBasicAuth],
+  [:session,            Warden::Strategies::Session]
+]
+
+strategies.each do |name, clazz|
+  Warden::Strategies.add name, clazz
+end
+
+include OpenProject::Authentication::Scope
+
+OpenProject::Authentication.update_strategies(API_V3) do |_strategies|
```

Comment:
Using an included const like that is something I've never seen before.
It does not prevent passing an invalid / unknown scope to `update_strategies` (with a symbol) and using :api_v3 instead would spare you the include.

Is that a common style I've missed elsewhere or does it serve another purpose I cannot see?